final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,83 @@
+import argparse
+from typing import Union
+import feedparser
+import logging
+import json
+import sys
+from rss_reader.work_with_file import read_feed_form_file
+from rss_reader.work_with_file import add_feed_to_file
+from rss_reader.work_with_text import get_string_with_result
+from rss_reader.work_with_json import to_json
+from rss_reader.work_with_json import limited_json
+from rss_reader.decorators import functions_log
+from rss_reader import __version__
+
+
+@functions_log
+def get_object_feed(url: str) -> Union[str, feedparser.FeedParserDict]:
+    try:
+        data = feedparser.parse(url)
+        if data.status == 200:
+            if data.bozo:
+                return f'ERROR: There is no rss feed at this url: {url}'
+            else:
+                return data
+        else:
+            return f'HTTP Status Code {data.status}'
+    except AttributeError:
+        return f'ERROR: {url} - is not url(example url "https://google.com")'
+    except Exception as exc:
+        return f'ERROR: {exc}'
+
+
+def set_start_setting():
+    """setup start settings"""
+    parser = argparse.ArgumentParser()
+    parser.add_argument("source", help="RSS URL", nargs='?', default='', type=str)
+    parser.add_argument("--version", help="Print version info", action="store_true")
+    parser.add_argument("--json", help="Print result as JSON in stdout", action="store_true")
+    parser.add_argument("--verbose", help="Outputs verbose status messages", action="store_true")
+    parser.add_argument("--limit", help="Limit news topics if this parameter provided", type=int)
+    parser.add_argument("--date", help="Obtaining the cached news without the Internet", type=str)
+    args = parser.parse_args()
+    if not args.limit:
+        args.limit = 1000
```

Comment:
And what if there will be 1001 entry in the feed? If user did not specify the limit it is better to link it somehow with feed size